### PR TITLE
Avoid name collision with parent method

### LINKED ISSUE #192: crash about getClient since sf 4.3.0
#### Symfony packages

```
symfony/asset                      v4.3.2  v4.3.2  Symfony Asset Component
symfony/browser-kit                v4.3.2  v4.3.2  Symfony BrowserKit Component
symfony/cache                      v4.3.2  v4.3.2  Symfony Cache component with PSR-6, PSR-16, and tags
symfony/cache-contracts            v1.1.5  v1.1.5  Generic abstractions related to caching
symfony/config                     v4.3.2  v4.3.2  Symfony Config Component
symfony/console                    v4.3.2  v4.3.2  Symfony Console Component
symfony/css-selector               v4.3.2  v4.3.2  Symfony CssSelector Component
symfony/debug                      v4.3.2  v4.3.2  Symfony Debug Component
symfony/dependency-injection       v4.3.2  v4.3.2  Symfony DependencyInjection Component
symfony/doctrine-bridge            v4.3.2  v4.3.2  Symfony Doctrine Bridge
symfony/dom-crawler                v4.3.2  v4.3.2  Symfony DomCrawler Component
symfony/event-dispatcher           v4.3.2  v4.3.2  Symfony EventDispatcher Component
symfony/event-dispatcher-contracts v1.1.5  v1.1.5  Generic abstractions related to dispatching event
symfony/expression-language        v4.3.2  v4.3.2  Symfony ExpressionLanguage Component
symfony/filesystem                 v4.3.2  v4.3.2  Symfony Filesystem Component
symfony/finder                     v4.3.2  v4.3.2  Symfony Finder Component
symfony/form                       v4.3.2  v4.3.2  Symfony Form Component
symfony/framework-bundle           v4.3.2  v4.3.2  Symfony FrameworkBundle
symfony/http-foundation            v4.3.2  v4.3.2  Symfony HttpFoundation Component
symfony/http-kernel                v4.3.2  v4.3.2  Symfony HttpKernel Component
symfony/inflector                  v4.3.2  v4.3.2  Symfony Inflector Component
symfony/intl                       v4.3.2  v4.3.2  A PHP replacement layer for the C intl extension that includes additional data from the ICU library.
symfony/mime                       v4.3.2  v4.3.2  A library to manipulate MIME messages
symfony/monolog-bridge             v4.3.2  v4.3.2  Symfony Monolog Bridge
symfony/monolog-bundle             v3.4.0  v3.4.0  Symfony MonologBundle
symfony/options-resolver           v4.3.2  v4.3.2  Symfony OptionsResolver Component
symfony/phpunit-bridge             v4.3.2  v4.3.2  Symfony PHPUnit Bridge
symfony/polyfill-ctype             v1.11.0 v1.11.0 Symfony polyfill for ctype functions
symfony/polyfill-intl-icu          v1.11.0 v1.11.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn          v1.11.0 v1.11.0 Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-mbstring          v1.11.0 v1.11.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72             v1.11.0 v1.11.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73             v1.11.0 v1.11.0 Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/process                    v4.3.2  v4.3.2  Symfony Process Component
symfony/property-access            v4.3.2  v4.3.2  Symfony PropertyAccess Component
symfony/routing                    v4.3.2  v4.3.2  Symfony Routing Component
symfony/security-acl               v3.0.2  v3.0.2  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle            v4.3.2  v4.3.2  Symfony SecurityBundle
symfony/security-core              v4.3.2  v4.3.2  Symfony Security Component - Core Library
symfony/security-csrf              v4.3.2  v4.3.2  Symfony Security Component - CSRF Library
symfony/security-guard             v4.3.2  v4.3.2  Symfony Security Component - Guard
symfony/security-http              v4.3.2  v4.3.2  Symfony Security Component - HTTP Integration
symfony/serializer                 v4.3.2  v4.3.2  Symfony Serializer Component
symfony/service-contracts          v1.1.5  v1.1.5  Generic abstractions related to writing services
symfony/templating                 v4.3.2  v4.3.2  Symfony Templating Component
symfony/translation                v4.3.2  v4.3.2  Symfony Translation Component
symfony/translation-contracts      v1.1.5  v1.1.5  Generic abstractions related to translation
symfony/twig-bridge                v4.3.2  v4.3.2  Symfony Twig Bridge
symfony/twig-bundle                v4.3.2  v4.3.2  Symfony TwigBundle
symfony/validator                  v4.3.2  v4.3.2  Symfony Validator Component
symfony/var-exporter               v4.3.2  v4.3.2  A blend of var_export() + serialize() to turn any serializable data structure to plain PHP code
symfony/yaml                       v4.3.2  v4.3.2  Symfony Yaml Component

```

#### Symfony CMF packages

```
symfony-cmf/resource             1.1.0  1.1.0  Bundle which facilitates document resource location via Puli
symfony-cmf/resource-bundle      1.1.0  1.1.0  Bundle which facilitates document resource location
symfony-cmf/resource-rest-bundle 1.1.0  1.1.0  Bundle which provides a REST API for resources
symfony-cmf/testing              2.1.11 2.1.11 Component providing tools for writing tests with Symfony.
symfony-cmf/tree-browser-bundle  2.1.1  2.1.1  Symfony CMF Tree Browser Bundle
```

## Subject

Since https://github.com/symfony/symfony/commit/4f91020c8df64b6e4609c9ee1a317789a4498cdd , that ships with 4.3.0, there is a collision between `    private static function getClient(KernelBrowser $newClient = null): ?KernelBrowser` https://github.com/symfony/symfony/blob/950306adaad8b23725c4efeaf6c570d89e17f164/src/Symfony/Bundle/FrameworkBundle/Test/WebTestAssertionsTrait.php#L189 and https://github.com/symfony-cmf/Testing/blob/a50228dd662c3104214049f61d68c1b325a60812/src/Functional/BaseTestCase.php#L62

## Steps to reproduce
Retry this job: run what this job runs: https://travis-ci.org/sonata-project/SonataDoctrinePhpcrAdminBundle/jobs/556144893#L656

## Expected results
No crash

## Actual results
A crash

### PR DESCRIPTION
**Pull request by greg0ire**
Since symfony/symfony@4f91020 , that ships with 4.3.0, there is a
collision between a private static method in a trait use in an ancestor
class of BaseTestCase and the method in BaseTestCase.
See https://github.com/symfony/symfony/blob/950306adaad8b23725c4efeaf6c570d89e17f164/src/Symfony/Bundle/FrameworkBundle/Test/WebTestAssertionsTrait.php#L189
Closes #192

| Q             | A
| ------------- | ---
| Branch?       | 2.1
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Fixed tickets | #192
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Comment by greg0ire** [comment]
Travis has RAM issues with the build it seems. What should we do?

File: phpunit.xml.dist [hunk 1]
```diff
@@ -12,4 +12,8 @@
         </testsuite>
     </testsuites>
 
+    <listeners>
+       <listener class="Symfony\Bridge\PhpUnit\SymfonyTestsListener" />
```
Comment (dbu) [inline phpunit.xml.dist]: is this related?

File: tests/Functional/BaseTestCaseTest.php [hunk 2]
```diff
@@ -76,6 +76,20 @@ public function provideTestDb()
         ];
     }
 
+    /**
+     * @group legacy
+     * @expectedDeprecation "Symfony\Cmf\Component\Testing\Functional\BaseTestCase::getClient()" is deprecated in favor of getFrameworkBundleClient() since symfony-cmf/testing 2.1 and will no longer be callable in 3.0
+     */
+    public function testItTriggersADeprecationErrorWhenCallingGetClient()
+    {
+        $this->testCase->getClient();
```
Comment (dbu) [inline tests/Functional/BaseTestCaseTest.php]: we should still assert that we get a Client::class with the old call.

File: phpunit.xml.dist (hunk 1 above)
Comment (greg0ire) [reply phpunit.xml.dist]: Yes, you cannot use assertions on the deprecations without this.

Comment (dbu) [reply phpunit.xml.dist]: ah, ok. cool !

File: src/Functional/BaseTestCase.php [hunk 3]
```diff
@@ -54,12 +54,23 @@ protected function getKernelConfiguration()
...
 
+    public function __call($name, $arguments)
+    {
+        if ('getClient' !== $name) {
+            return call_user_func_array($name, $arguments);
+        }
+        @trigger_error(sprintf(
+            '"%s::getClient()" is deprecated in favor of getFrameworkBundleClient() since symfony-cmf/testing 2.1 and will no longer be callable in 3.0',
+            __CLASS__
+        ), E_USER_DEPRECATED);
```
[the remark this answers is not in the document]
Comment (greg0ire) [reply src/Functional/BaseTestCase.php]: Wow. What was I thinking when I wrote this?

**Comment by greg0ire** [comment]
@dbu please review again

**Comment by dbu** [comment]
thanks!

**Comment by dbu** [comment]
created 2.1.12 with this, now i will merge to master and remove the deprecated part.
